Refactoring into service

NoDiskInDriveA opened the pull request:
There are multiple usecases that demand lookup of IVW properties in a different context than the current route. To allow this, I have refactored the module functions responsible for the hierarchical lookup into a service (using the Hierarchical Config module as an example) with additional abilities; namely lookup by any route, node or term.

I also removed duplicated code (lookup function in widget class) and some unused service injections.

Commit:
Removed unused contructor injections

Commit:
Implemented lookup functionality in service

Commit:
Removed module lookup code and replaced by service

Commit:
Fixed service injection parameters

Commit:
Added $parentsOnly param to node lookups
- parent lookup in settings field should now work correctly

Commit:
Corrected formatting

Commit:
Rearranged methods
Changed visibility of searchTerm()
Added documentation and type hints to interface
Renamed $parentsOnly to $termsOnly

dbosen commented:
The way $termsOnlyx flag is thrown aroiund is a bit ugly, but otherwise looks good to me. But I will test this with instyle before merging the request

NoDiskInDriveA commented:
Yeah I didn't like that either, $termsOnly was mainly an afterthought to quickly remove lookup code from the widget class. I think I could easily get rid of it by restructuring the methods a bit better.

The question is, how long will this module still be in use? When will it be replaced by Hierarchical Config? (btw. I think HC should get a similar interface...)

Commit:
INREL-710: add code to this module (for experimentation)

Commit:
INREL-710: saving changes

Commit:
Moved FIA stuff to submodule
Added config for FIA domain
Removed experimental code

Commit:
Fix getting the referencing entity for the IVW field (seems to be a regression)

Commit:
Remove todo

Commit:
- fix parent lookup when examined entity is a term
- rename $termsOnly to $parentsOnly (it concerns all kinds of entities now)
- add $parentsOnly to more functions (terms need it too)
- documentation

Commit:
Move out fia functionality to external repo

NoDiskInDriveA commented:
I fixed some lookup logic and regression bugs that occur with Drupal 8.1.7.

There now seems to be an empty term entity reference field on every term so the parent lookup logic (which basically hadn't been modified) sometimes fails b/c it thinks it's on a node instead of a term. As said, this seems to concern the base branch too (we noticed that with our relaunch of freundin.de). This is fixed in this PR now.

Also I rebased my fork to the current base HEAD so the merge would be a fast forward again.

Commit:
- fix config schema
